packages/google_maps_flutter/google_maps_flutter/example/integration_test/src/tiles_inspector.dart

```diff
@@ -208,269 +207,574 @@ void runTests() {
+      final Completer<int> mapIdCompleter = Completer<int>();
+      final Key key = GlobalKey();
+
+      await tester.pumpWidget(
+        Directionality(
+          textDirection: TextDirection.ltr,
+          child: GoogleMap(
+            key: key,
+            initialCameraPosition: kInitialCameraPosition,
+            heatmaps: <Heatmap>{heatmap1},
+            onMapCreated: (GoogleMapController controller) {
+              mapIdCompleter.complete(controller.mapId);
+            },
+          ),
         ),
-        HeatmapGradientColor(
-          Color.fromARGB(255, 0, 0, 191),
-          0.6,
+      );
+      await tester.pumpAndSettle(const Duration(seconds: 3));
+
+      final int mapId = await mapIdCompleter.future;
+      final GoogleMapsInspectorPlatform inspector =
+          GoogleMapsInspectorPlatform.instance!;
+
+      final Heatmap heatmap1New = heatmap1.copyWith(
+        dataParam: heatmap1.data.sublist(5),
+        dissipatingParam: !heatmap1.dissipating,
+        gradientParam: heatmap1.gradient,
+        maxIntensityParam: heatmap1.maxIntensity! + 1,
+        opacityParam: heatmap1.opacity - 0.1,
+        radiusParam: HeatmapRadius.fromPixels(heatmap1.radius.radius + 1),
+        minimumZoomIntensityParam: heatmap1.minimumZoomIntensity + 1,
+        maximumZoomIntensityParam: heatmap1.maximumZoomIntensity + 1,
+      );
+
+      await tester.pumpWidget(
+        Directionality(
+          textDirection: TextDirection.ltr,
+          child: GoogleMap(
+            key: key,
+            initialCameraPosition: kInitialCameraPosition,
+            heatmaps: <Heatmap>{heatmap1New},
+            onMapCreated: (GoogleMapController controller) {
+              fail('update: OnMapCreated should get called only once.');
+            },
+          ),
         ),
-        HeatmapGradientColor(
-          Color.fromARGB(255, 63, 0, 91),
-          0.8,
+      );
+
+      await tester.pumpAndSettle(const Duration(seconds: 3));
+
+      if (inspector.supportsGettingHeatmapInfo()) {
+        final Heatmap heatmapInfo1 =
+            (await inspector.getHeatmapInfo(heatmap1.mapsId, mapId: mapId))!;
+
+        expectHeatmapEquals(heatmap1New, heatmapInfo1);
+      }
+    });
+
+    testWidgets('remove heatmaps correctly', (WidgetTester tester) async {
+      final Completer<int> mapIdCompleter = Completer<int>();
+      final Key key = GlobalKey();
+
+      await tester.pumpWidget(
+        Directionality(
+          textDirection: TextDirection.ltr,
+          child: GoogleMap(
+            key: key,
+            initialCameraPosition: kInitialCameraPosition,
+            heatmaps: <Heatmap>{heatmap1},
+            onMapCreated: (GoogleMapController controller) {
+              mapIdCompleter.complete(controller.mapId);
+            },
+          ),
         ),
-        HeatmapGradientColor(
-          Color.fromARGB(255, 255, 0, 0),
-          1,
+      );
+      await tester.pumpAndSettle(const Duration(seconds: 3));
```

Comment:
Without this line heatmap tests timeouted on Web platform, and we were not able to test ground overlay below.

Reply:
(Same as above, resolving)